omnilib/control.lua

```diff
@@ -1,408 +1,420 @@
 require("controlFunctions")
+--
+    --
+        --
+            -- 
+                --
+                    --
+                        --
+                            --
+                                --
+                                    -- RIP MT. OMNISSIAH
+                                --
+                            --
+                        --
+                    --
+                --
+            --
+        --
+    --
+--
 
-local omni = {tech = {},rec={}}
-
-
-
+local function get_recipe_family(recipe)
+	local metadata = global.omni.recipes[recipe.name]
+	if not metadata then
+		return
+	end
+	local recipe_tree = global.omni.force_recipes[recipe.force.name][metadata.base]
+	if not recipe_tree then
+		return
+	end
+	return recipe_tree, metadata
+end
 
-script.on_event(defines.events.on_player_joined_game, function(event)
+local function update_last_tier(recipe, full_refresh)
+	if not recipe.valid or not recipe.enabled then
+		return
+	end
+	full_refresh = full_refresh or global.omni.need_update
+	local recipe_tree, metadata = get_recipe_family(recipe)
+	if not recipe_tree then
+		return
+	end
+	-- We're the top of the tree!
+	if recipe_tree and metadata.variant == "" and recipe_tree.active_tier < metadata.tier then
+		local I = full_refresh and 97 or recipe_tree.active_tier
+		local research_status = not not (recipe.force.technologies["compression-recipes"] or {}).researched
```

Comment:
not not?

Reply:
"not not" is equivalent in usage to a boolean cast